CV Backend/src/main/java/com/qa/demo/Interoperability/AccountController.java

```diff
@@ -0,0 +1,233 @@
+                        return "File stream could not be closed";
+                        }
+                    }
+                }
+            }
+        }
+        return "File retrieved successfully";
+    }
+
+    @GetMapping(value = "/{id}")
+    public Account getAccountById(@PathVariable("id") ObjectId id){
+        return repository.findBy_id(id);
+    }
+
+    @GetMapping(value = "/role/{role}")
+    public List<Account> getAccountByRole(@PathVariable("role") String search){
+        return repository.findByUserRole(search);
+    }
+
+    @PostMapping(value = "/mail/{email}")
+    public Account getAccountByEmail(@PathVariable("email") String email, HttpServletResponse response){
+        Account retrieved = repository.findByEmail(email);
+        if (retrieved == null){
+            response.setStatus(404);}
+        return retrieved;
+    }
+
+    @PutMapping(value = "/{id}")
+    public String modifyAccountById(Principal principal, @PathVariable("id")ObjectId id, @Valid @RequestBody Account account){
+        if(validOperation(principal, id)) {
+            Account updatable = repository.findBy_id(id);
+            account.set_id(id);
+            updatable.updateFields(account);
+            repository.save(updatable);
+            return "Account Updated";
+        }
+        return "Not Authorised";
+    }
+    
+    @PutMapping(value = "/password/{id}")
+    public String modifyPasswordById(Principal principal, @PathVariable("id")ObjectId id, @RequestBody PasswordObject passwords, HttpServletResponse response){
+        Account updatable = repository.findBy_id(id);
+        if(validOperation(principal, id) && bCode.matches(passwords.getOldPassword(), updatable.getPassword())) {
+            updatable.setPassword(bCode.encode(passwords.getNewPassword()));
+            repository.save(updatable);
+            return "Account Updated";
+        }
+        else{
+            response.setStatus(404);
+        }
+        return "Not Authorised";
+    }
+
+
+    @DeleteMapping(value = "/{id}")
+    public String deleteAccount(Principal principal, @PathVariable ObjectId id){
+        if (validOperation(principal, id)) {
+            repository.delete(repository.findBy_id(id));
+            return "Account Deleted";
+        }
+        return "Not Authorised";
+    }
+
+    @DeleteMapping(value = "/deletecv/{id}")
+    public String deleteCV(Principal principal, @PathVariable ObjectId id){
+        if(validOperation(principal, id)){
+            mongoTemplate.upsert(Query.query(where("_id").is(id)), Update.update("file",null), Account.class);
+            return "CV was deleted";
+        }
+        return "CV could not be deleted";
+    }
+
+    private boolean validOperation(Principal principal, ObjectId id){
+        Account accountOfCall = repository.findByEmail(principal.getName());
+        return (accountOfCall.getUserRole().equalsIgnoreCase("admin") || accountOfCall.get_id().equals(id.toHexString())) ? true : false;
```

Comment:
(accountOfCall.getUserRole().equalsIgnoreCase("admin") || accountOfCall.get_id().equals(id.toHexString())) is by itself returning a boolean for you therefore you dont need the "? true : false"

Reply:
Fixed